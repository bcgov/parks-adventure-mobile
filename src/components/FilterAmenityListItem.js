import React from 'react'
import PropTypes from 'prop-types'
import { IconButton } from 'react-native-paper'
import { getParkSvg } from '../images/svgs'
import { Container, Title, Selection } from './FilterAmenityListItem.styles'

const FilterAmenityListItem = ({ name, onSelect, selected }) => (
  <Container>
    {getParkSvg(name, selected)}
    <Title selected={selected}>{name}</Title>
    <Selection selected={selected} accessibilityLabel={'select'}>
      <IconButton icon="check" color="white" size={26} onPress={onSelect} />
    </Selection>
  </Container>
)

FilterAmenityListItem.propTypes = {
  name: PropTypes.string.isRequired,
  onSelect: PropTypes.func.isRequired,
  selected: PropTypes.bool,
}

export default FilterAmenityListItem
