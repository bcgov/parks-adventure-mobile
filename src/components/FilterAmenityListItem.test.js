import React from 'react'
import { render } from '@testing-library/react-native'
import renderSnapshot from 'react-test-renderer'
import FilterAmenityListItem from './FilterAmenityListItem'

test('Should render as selected', () => {
  const { getByText, getByA11yLabel } = render(
    <FilterAmenityListItem name="Hiking" onSelect={() => {}} selected={true} />
  )

  const title = getByText('Hiking')
  expect(title).toBeDefined()

  const checked = getByA11yLabel('select')
  expect(checked).toBeDefined()
  expect(checked._fiber.stateNode.props.style[0].backgroundColor).toBe(
    '#294071'
  )
  expect(checked._fiber.stateNode.props.style[0].borderColor).toBe('#294071')
})

test('Should render as unselected', () => {
  const { getByText, getByA11yLabel } = render(
    <FilterAmenityListItem name="Hiking" onSelect={() => {}} selected={false} />
  )

  const title = getByText('Hiking')
  expect(title).toBeDefined()

  const checked = getByA11yLabel('select')
  expect(checked).toBeDefined()
  expect(checked._fiber.stateNode.props.style[0].backgroundColor).toBe('white')
  expect(checked._fiber.stateNode.props.style[0].borderColor).toBe('black')
})

test('matches snapshot', () => {
  const tree = renderSnapshot
    .create(
      <FilterAmenityListItem
        name="Hiking"
        onSelect={() => {}}
        selected={true}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
