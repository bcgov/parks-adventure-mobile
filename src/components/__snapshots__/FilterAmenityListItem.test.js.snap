// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 8,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 0,
      },
    ]
  }
>
  <SvgMock
    height={36}
    width={36}
  />
  <Text
    selected={true}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "black",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 18,
            "lineHeight": 23,
            "marginLeft": 30,
          },
        ],
      ]
    }
  >
    Hiking
  </Text>
  <View
    accessibilityLabel="select"
    selected={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#294071",
          "borderColor": "#294071",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 2,
          "height": 28,
          "justifyContent": "center",
          "width": 28,
        },
      ]
    }
  >
    <Icon
      color="white"
      icon="check"
      onPress={[Function]}
      size={26}
    />
  </View>
</View>
`;
